components/vc/shared/code/HW_tim_vcSpecific.c

```diff
@@ -0,0 +1,144 @@
+/**
+ * @file HW_tim.c
+ * @brief  Source code for TIM firmware
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+// Firmware Includes
+#include "HW_tim.h"
+
+/******************************************************************************
+ *                              D E F I N E S
+ ******************************************************************************/
+
+#define TIMER_BASE_TICK 1000000U
+#define SEC_TO_BASETICK(s) ((s) * TIMER_BASE_TICK)
+
+#define TICK_PER_REV 16U
+#define UPDATE_PER_REV 4U
+
+#define NUM_TIM_SAMPLES 2U
+#define CURRENT_SAMPLE (NUM_TIM_SAMPLES - 1U)
+#define LAST_SAMPLE (NUM_TIM_SAMPLES - 2U)
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+TIM_HandleTypeDef htim[HW_TIM_PORT_COUNT];
+typedef struct
+{
+    uint64_t lastSample[HW_TIM_CHANNEL_WS_CNT][NUM_TIM_SAMPLES];
+} wheelSpeed_data_S;
+
+static wheelSpeed_data_S wheelSpeed;
```

Comment:
god the body controls coding is so strong in you lmao, this is the kind of stuff I don't like about how body controls writes their code

this is a private variable, it will only ever be accessed within the context of this file. `wheelSpeed_data_S` is not any more clear than `ws_S` when you're in the context of this file, but it is longer and more annoying. It's not like you have to jump to a different file to see what it means, plus with modern IDE completion you get it for free with `Shift+k` or hover. Same for `wheelSpeed` vs just `WS`.